Intersect.Client/Core/Audio.cs

```diff
@@ -283,6 +283,15 @@ public static void StopSound(IMapSound sound)
             sound?.Stop();
         }
 
+        public static void StopAllGameSoundsOf(string[] filenames)
+        {
+            var validSounds = sGameSounds.Where(s => filenames.Contains(s.Filename)).ToArray();
```

Comment:
I think you can change it to Where(filenames.Contains)?

Reply:
Also don't bother with the ToArray(), just leave it as IEnumerable<string>

Reply:
I don't think we can do that - a sound contains more than just its filename. I could do a select to get the filenames, and then a where like you suggested _after_ that, but that sounds wasteful